WcaOnRails/app/controllers/payment_controller.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+class PaymentController < ApplicationController
+  def payment_config
+    payment_id = params.require(:payment_id)
+    competition_id = params.require(:competition_id)
+
+    competition = Competition.find(competition_id)
+    stripe_transaction = StripeTransaction.find(payment_id)
+    secret = stripe_transaction.stripe_payment_intent.client_secret
+    render json: { stripe_publishable_key: AppSecrets.STRIPE_PUBLISHABLE_KEY,
```

Comment:
I presume the Reg microservice needs this to render the Stripe UI. Why can't you store this in Vault and have the Reg service access it directly from there?

Reply:
I was following the stripe example on their github https://github.com/stripe-samples/accept-a-payment/blob/main/payment-element/client/react-cra/src/App.js
where they also have a config route to read the key. Regardless if the registration service or the payment service saves them in vault, we still need to use the config route to fetch it